src/main/java/com/aws/greengrass/clientdevices/auth/certificate/listeners/ConfiguredCertificateAuthorityEventListener.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.listeners;
+
+import com.aws.greengrass.clientdevices.auth.api.DomainEvents;
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.certificate.events.ConfiguredCertificateAuthorityEvent;
+import com.aws.greengrass.clientdevices.auth.certificate.usecases.RegisterCertificateAuthorityUseCase;
+import com.aws.greengrass.clientdevices.auth.exception.UseCaseException;
+
+import javax.inject.Inject;
+
+public class ConfiguredCertificateAuthorityEventListener {
```

Comment:
Could we use the `DomainEventListener` interface?. Also I am really happy you added a Listener class which can serve as a controller to redirect to the proper use case based on the event contents